engine/src/main/java/org/entando/entando/aps/system/init/InitializerManager.java

```diff
@@ -37,164 +37,164 @@
-			DataSource dataSource = (DataSource) this.getBeanFactory().getBean("portDataSource");
-			dao.setDataSource(dataSource);
-			dao.saveConfigItem(report.toXml(), this.getConfigVersion());
-			this.getCacheWrapper().setCurrentReport(report);
-		} catch (Throwable t) {
-			logger.error("Error saving report", t);
-			throw new FatalBeanException("Error saving report", t);
-		}
-	}
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    public static final String REPORT_CONFIG_ITEM = "entandoComponentsReport";
+
+    private IInitializerManagerCacheWrapper cacheWrapper;
+
+    private boolean checkOnStartup;
+
+    private Map<String, IPostProcessor> postProcessors;
+
+    private IDatabaseManager databaseManager;
+
+    @Override
+    public SystemInstallationReport getCurrentReport() {
+        return this.getCacheWrapper().getReport();
+    }
+
+    protected IInitializerManagerCacheWrapper getCacheWrapper() {
+        return cacheWrapper;
+    }
+
+    public void setCacheWrapper(IInitializerManagerCacheWrapper cacheWrapper) {
+        this.cacheWrapper = cacheWrapper;
+    }
+
+    protected boolean isCheckOnStartup() {
+        return checkOnStartup;
+    }
+
+    public void setCheckOnStartup(boolean checkOnStartup) {
+        this.checkOnStartup = checkOnStartup;
+    }
+
+    public Map<String, IPostProcessor> getPostProcessors() {
+        return postProcessors;
+    }
+
+    public void setPostProcessors(Map<String, IPostProcessor> postProcessors) {
+        this.postProcessors = postProcessors;
+    }
+
+    protected IDatabaseManager getDatabaseManager() {
+        return databaseManager;
+    }
+
+    public void setDatabaseManager(IDatabaseManager databaseManager) {
+        this.databaseManager = databaseManager;
+    }
+
+    public void init() throws Exception {
+        SystemInstallationReport report = null;
+        try {
+            report = this.extractReport();
+            report = ((IDatabaseInstallerManager) this.getDatabaseManager()).installDatabase(report, this.isCheckOnStartup());
+            this.getCacheWrapper().initCache(report);
+        } catch (Throwable t) {
+            logger.error("Error while initializating Db Installer", t);
+            throw new Exception("Error while initializating Db Installer", t);
+        } finally {
+            if (null != report && report.isUpdated()) {
+                this.saveReport(report);
+            }
+        }
+        logger.debug("{}: initializated - Check on startup {}", this.getClass().getName(), this.isCheckOnStartup());
+    }
+
+    public void executePostInitProcesses() throws BeansException {
+        SystemInstallationReport report = null;
+        try {
+            report = this.extractReport();
+            List<Component> components = this.getComponentManager().getCurrentComponents();
+            for (Component component : components) {
+                executeComponentPostInitProcesses(component, report);
```

Comment:
Best improvement of this changeset, very good

Reply:
thanks!